### PR TITLE
Fixing build/test errors

### PR DESCRIPTION
**Pull request by yongyan-gh**
# Description:

- Fix high severity vulnerability in SqlClient packages which break build. more information:  https://github.com/advisories/GHSA-98g6-xh36-x2p7

- Fix an issue cause StackTests fail if the stack trace contains relative file location.

### CODE REVIEW

**Review by stacywray — APPROVED** [review]
:shipit:

File: src/Test.UnitTests.Sarif/Core/StackTests.cs [hunk 1]
```diff
@@ -111,5 +111,44 @@ public void Stack_CreateFromAggregatedExceptionWithInnerException()
             }
             Assert.True(caughtException);
         }
+
+        [Fact]
+        public void Stack_CreateFromStackTraceWithRelativeSourceFileLocation()
```
Comment (michaelcfanning) [inline src/Test.UnitTests.Sarif/Core/StackTests.cs:116]: >Stack_CreateFromStackTraceWithRelativeSourceFileLocation [](http://example.com/codeflow?start=20&length=56)

This is a good fix, can we add a release notes entry please?

File: ReleaseHistory.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 ## UNRELEASED
 * DEP: Remove explicit versioning for `System.Memory` and `System.Runtime.CompilerServices.Unsafe`.
 * DEP: Remove spurious references to `System.Collections.Immutable`.
+* DEP: Update `Microsoft.Data.SqlClient` reference from 2.1.2 to 2.1.7 in `WorkItems` and `Sarif.Multitool.Library` to avoid [CVE-2024-0056](https://github.com/advisories/GHSA-98g6-xh36-x2p7) vulnerability.
```
[the remark this answers is not in the document]
Comment (yongyan-gh) [reply ReleaseHistory.md]: updated

File: src/Test.UnitTests.Sarif/Core/StackTests.cs (hunk 1 above)
Comment (yongyan-gh) [reply src/Test.UnitTests.Sarif/Core/StackTests.cs:116]: Added a release record

File: ReleaseHistory.md [hunk 3]
```diff
@@ -2,6 +2,9 @@
 ## UNRELEASED
 * DEP: Remove explicit versioning for `System.Memory` and `System.Runtime.CompilerServices.Unsafe`.
 * DEP: Remove spurious references to `System.Collections.Immutable`.
+* DEP: Update `Microsoft.Data.SqlClient` reference from 2.1.2 to 2.1.7 in `WorkItems` and `Sarif.Multitool.Library` to resolve [CVE-2024-0056](https://github.com/advisories/GHSA-98g6-xh36-x2p7).
+* DEP: Update `System.Data.SqlClient` reference from 4.8.5 to 4.8.6 in `WorkItems` to resolve [CVE-2024-0056](https://github.com/advisories/GHSA-98g6-xh36-x2p7).
+* BUG: Fix the issue `Stack` object doesn't create corresponding `PhysicalLocation` from stack trace if it contains relative file path.
```
Comment (michaelcfanning) [inline ReleaseHistory.md:7]: >' [](http://example.com/codeflow?start=41&length=1)

Your release note doesn't include information on what helper was literally updated. Please include this information.

Your method is `StackFrame.Create`. Please try to provide this level of specificity for a change like this. You have told customers 'i updated the stack object' but a stronger statement is 'i fixed this method on this class'. 

> * BUG: Update `Stack.Create` method to properly create `PhysicalLocation` instances when stack frames reference relative file paths.

**Review by shaopeng-gh — APPROVED** [review]
:shipit:

File: ReleaseHistory.md (hunk 3 above)
Comment (yongyan-gh) [reply ReleaseHistory.md]: updated according to offline chat.

**Pull request merged**